backend/tests/plan/test_primary_schedules.py

```diff
@@ -0,0 +1,145 @@
+        # self.assertEqual(response.data["id"], self.s2.id)
+
+    def test_primary_schedule_friends(self):
+        response = self.client.put(primary_schedule_url, {"schedule_id": self.s.id})
+
+        u2 = User.objects.create_user(
+            username="jacob2", email="jacob2@gmail.com", password="top_secret"
+        )
+        u3 = User.objects.create_user(
+            username="jacob3", email="jacob3@gmail.com", password="top_secret"
+        )
+        self.client2 = APIClient()
+        self.client2.login(username="jacob2", password="top_secret")
+        self.client3 = APIClient()
+        self.client3.login(username="jacob3", password="top_secret")
+
+        Friendship.objects.create(sender=self.u1, recipient=u2, status=Friendship.Status.ACCEPTED)
+        u2_s = Schedule(
+            person=u2,
+            semester=TEST_SEMESTER,
+            name="U2 Test Schedule",
+        )
+        u2_s.save()
+        u2_s.sections.set([self.cis120])
+        response = self.client2.put(primary_schedule_url, {"schedule_id": u2_s.id})
+        # self.assertEqual(response.status_code, 200)
+        # self.assertEqual(response.data["id"], u2_s.id)
+
+        response = self.client.get(primary_schedule_url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.json()), 2)
+        # print("1", response.json())
+        # self.assertEqual(response.data[0]["id"], self.s.id)
+        # self.assertEqual(response.data[1]["id"], u2_s.id)
+
+        Friendship.objects.create(sender=self.u1, recipient=u3, status=Friendship.Status.ACCEPTED)
+        u3_s = Schedule(
+            person=u3,
+            semester=TEST_SEMESTER,
+            name="U3 Test Schedule",
+        )
+        u3_s.save()
+        u3_s.sections.set([self.cis120])
+
+        # shouldn't change bc no primary scheudle for u3 yet
+        response = self.client.get(primary_schedule_url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.json()), 2)
+        # print("2", response.json())
+
+        # add a primary schedule for u3
+        response = self.client3.put(primary_schedule_url, {"schedule_id": u3_s.id})
+        self.assertEqual(response.status_code, 200)
+        # self.assertEqual(response.data["id"], u3_s.id)
+
+        # should have all 3 now
+        response = self.client.get(primary_schedule_url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.json()), 3)
+        # print("3", response.json())
+
+        # remove u2 as a friend
+        friendshipu2 = Friendship.objects.get(sender=self.u1, recipient=u2)
+        friendshipu2.delete()
+
+        # only have u1 and u3 now
+        response = self.client.get(primary_schedule_url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.json()), 2)
+        # print("4", response.json())
+        # self.assertEqual(response.data[0]["id"], self.s.id)
```

Comment:
I checked and wasn't able to remember, but I think it has to do with the schedules not loading properly in the testing environment (so all of the primary schedule tests were not set up correctly). It might be better if we just directly test out the endpoint since it's only 2 methods anyways.